types/jss/index.d.ts

```diff
@@ -3,7 +3,187 @@
 // Definitions by: Brenton Simpson <https://github.com/appsforartists>
 //                 Oleg Slobodskoi <https://github.com/kof>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 2.3
+
+import { CSSProperties } from 'react';
+
+export type PseudoCssKey =
+	| ':active'
+	| ':any'
+	| ':checked'
+	| ':default'
+	| ':disabled'
+	| ':empty'
+	| ':enabled'
+	| ':first'
+	| ':first-child'
+	| ':first-of-type'
+	| ':fullscreen'
+	| ':focus'
+	| ':hover'
+	| ':indeterminate'
+	| ':in-range'
+	| ':invalid'
+	| ':last-child'
+	| ':last-of-type'
+	| ':left'
+	| ':link'
+	| ':only-child'
+	| ':only-of-type'
+	| ':optional'
+	| ':out-of-range'
+	| ':read-only'
+	| ':read-write'
+	| ':required'
+	| ':right'
+	| ':root'
+	| ':scope'
+	| ':target'
+	| ':valid'
+	| ':visited'
+	// TODO
+	// | ':dir()'
+	// | ':lang()'
+	// | ':not()'
+	// | ':nth-child()'
+	// | ':nth-last-child()'
+	// | ':nth-last-of-type()'
+	// | ':nth-of-type()'
+	| '::after'
+	| '::before'
+	| '::cue'
+	| '::first-letter'
+	| '::first-line'
+	| '::selection'
+	| '::backdrop '
+	| '::placeholder '
+	| '::marker '
+	| '::spelling-error '
+	| '::grammar-error ';
+
+export type PseudoCss = Partial<Record<PseudoCssKey, CSSProperties>>;
+
+export interface JssProps {
+	'@global'?: React.CSSProperties & PseudoCss;
+	extend?: string;
+	composes?: string | string[];
+}
+
+export type css = React.CSSProperties;
+
+export interface JssExpand {
+	animation:
+		| {
```

Comment:
I've never seen a type declaration start with a `|`.

I can guess what the `delay: css['animationDelay']` does but I'm curious to see a comment or link explaining the leading `|`.

Reply:
The leading `|` has no significance, it's purely for uniformity of presentation, just like trailing commas in JS 🙂